device_backends/include/IEEE754_SingleConverter.h

```diff
@@ -32,8 +32,10 @@ namespace ChimeraTK {
     template<typename CookedType, typename RAW_ITERATOR, typename COOKED_ITERATOR>
     void vectorToCooked(
         const RAW_ITERATOR& raw_begin, const RAW_ITERATOR& raw_end, const COOKED_ITERATOR& cooked_begin) const {
-      static_assert(std::is_same<typename std::iterator_traits<RAW_ITERATOR>::value_type, int32_t>::value,
-          "RAW_ITERATOR template argument must be an iterator with value type equal to int32_t.");
+      static_assert(std::is_same<typename std::iterator_traits<RAW_ITERATOR>::value_type, int8_t>::value ||
```

Comment:
The static_assert is about class instantiation, not about object instantiation (i.e. actual use). In our current abstraction model we need to instantiate the class for all raw types (see the new `callForRawTypes` function I introduced in `SupportedUserTypes.h`). We could avoid this at the cost of needing to maintain the list of acceptable raw types at many places.

We do not have the notion of a "transport layer raw" in the framework, this is already abstracted by the backend (well, we now have some information about it through `minimumTransferAlignment`). The "before shape conversion raw" is essentially only internally visible in the `LowLevelTransferElement` and is fixed to `int8_t`. The only raw type that is visible to the outside works it the "before conversion raw". The separation is hence actually quite clear in the framework (but maybe not to us and the users, as we have not though about this that way before).

Reply:
(Btw: IEEE 754-2008, the current version of IEEE 754, contains also 16 bit floats, and we might well write a converter for that - it just wouldn't rely on a reinterpret_cast.)

Reply:
OK, I understand that in callForRaw types the compiler has to be able to generate code for all raw types without running into the static assertion. I still find it confusing, because it breaks locality of the code. This is a converter for single precision IEEE754 (32 bit), hence the name.
But I understand that a static assertion can be helpful here, but I would put a code comment. 

How about this:
* Put a code comment "// converter classes must be instantiable (i.e. the compiler must be able to generate the code without running into a compile time assertion) for all raw types that are supported, and only for those"

* Add a runtime check
```
// This converter is only working for int32_t. As the class code has to be instatiable for all raw types supported by ChimeraTK, we cannot have a compile time assertion about this. So we do a runtime check.
if (!std::is_same<typename std::iterator_traits<RAW_ITERATOR>::value_type, int32_t>::value){
  throw ChimeraTK::runtime_error("IEEE754_SingleConverter only works with raw type int32_t");
  }
```

Reply:
I would make it an assert. There is no way for a user to end up there. If someone manages to run into that situation, he has found a bug in DeviceAccess.

Reply:
Done in 564bbc9b70f6e4b97fcf44b7b608a8b24eeb1d0b (I just a slightly different wording).